packages/runtime/src/widgets/Search.tsx

```diff
@@ -52,16 +73,36 @@ export const Search: React.FC<SearchProps> = ({
               }))
           : [],
       );
+    } else if (onValueChange) {
+      (async (): Promise<void> => {
+        const apiResponse = await valueChangeAction?.callback();
+
+        if (apiResponse) {
+          setOptions(
+            findValuesByKey<string>(apiResponse, searchKey ?? "")
+              .filter((item) =>
+                item.toLowerCase().includes(search.toLowerCase()),
+              )
+              .map((item) => ({
+                value: item,
+                label: item,
+              })),
```

Comment:
This block should be in YAML instead of hardcoded to into the component. Expose setOptions as a setter.